### PR TITLE
feature(gce provision): support random zone

### PR DESCRIPTION
**Pull request by bentsi**
added ability to randomely select availability zone when not provided
https://trello.com/c/ZfJD4QDd/1900-support-random-azs-in-gce

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -386,15 +386,27 @@ def get_cluster_gce(self, loader_info, db_info, monitor_info):
...
         services = []
-        for i in gce_datacenter:
+        gce_datacenter = []
+        for region_az in regions:
+            # dc can be "us-east1-b" or "us-east1"
+            assert "us-east1" in region_az, "only 'us-east1' region is supported"
+            if region_az.count("-") == 1:
+                # randomly select zone and add it as a
+                # https://cloud.google.com/compute/docs/regions-zones#available
+                zone = random.choice(["c", "d"])
```
Comment (roydahan) [inline sdcm/tester.py:401]: What happened to a and b?

Comment (bentsi) [reply sdcm/tester.py:401]: there is no "a" zone in us-east1, only b,c,d. Check here: https://cloud.google.com/compute/docs/regions-zones#locations

**Comment by fruch** [comment]
@bentsi, sound like it actually was working, and worked on `us-east1-d`, be still got preempted:
```
SpotTerminationEvent Severity.CRITICAL): node=Node pr-provision-test-jenkins-db-node-d61c0abf-0-3 [35.231.148.216 | 10.142.0.98] (seed: False) message=Instance was preempted.
```

I think we'll need to think of a way to retry, if we want to keep using preempted

**Comment by bentsi** [comment]
@fruch it's better than nothing. 
I don't think we need to retry, the nature of spot/preemtible that they can get interrupted and terminated and to develop a mechanism that will recreate the instance on the fly during the test is too complex.

**Comment by fruch** [comment]
> @fruch it's better than nothing.
> I don't think we need to retry, the nature of spot/preemtible that they can get interrupted and terminated and to develop a mechanism that will recreate the instance on the fly during the test is too complex.

I was thinking more in re-triggering the whole job again.
thing is that we are spinning lot of those jobs at the same time, on gce. kind of shooting our self in leg, maybe we could find a way to trigger the same job in few hours.

**Comment by fruch** [comment]
I've added my thought into: https://trello.com/c/eOpLq9vN

**Pull request merged**